sql/core/src/test/scala/org/apache/spark/sql/RuntimeConfigSuite.scala

```diff
@@ -54,4 +54,15 @@ class RuntimeConfigSuite extends SparkFunSuite {
       conf.get("k1")
     }
   }
+
+  test("SPARK-24761: is a config parameter modifiable") {
+    val conf = newConf()
+
+    // SQL configs
+    assert(!conf.isModifiable("spark.sql.sources.schemaStringLengthThreshold"))
```

Comment:
nit: add a case for nonexistent sql conf key?

Reply:
Added a couple tests